app/src/main/kotlin/com/music/echo/ui/player/PlayerSyncedLyricsView.kt

```
@@ -0,0 +1,128 @@
+                    val existing = database.lyrics(mediaMetadata.id).firstOrNull()
+                    if (existing != null) return@launch
+                    val entryPoint = EntryPointAccessors.fromApplication(
+                        context.applicationContext,
+                        echo.music.iad1tya.di.LyricsHelperEntryPoint::class.java
+                    )
+                    val lyricsHelper = entryPoint.lyricsHelper()
+                    val fetchedLyricsWithProvider = lyricsHelper.getLyrics(mediaMetadata)
+                    database.query {
+                        upsert(LyricsEntity(mediaMetadata.id, fetchedLyricsWithProvider.lyrics ?: "", fetchedLyricsWithProvider.providerName))
+                    }
+                } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)